### DISCUSSION TITLE
Deploying Strimzi with custom storageClass

### DISCUSSION THREAD

**Discussion by davehouser1** [strimzi/strimzi-kafka-operator#4742] [Q&A]
I read though the quackster guide, which seems to be focused on Minikube and Kubernetes Kind. 
I am deploying on separate master / worker nodes, and use specific storageClasses for deployment. Following the guide fails as pods cant be scheduled due to using the default storage classes.

Do I literally need to download the yaml and edit, or is there an easier way?

**Comment by scholzj** [comment]
Well, the Quickstart is just a Quickstart. More details about the Storage configuration can be found in the Using guide for example. A custom storage class can be configured in the `class` option in the storage section: https://strimzi.io/docs/operators/latest/full/using.html#ref-persistent-storage-str

Hope this helps. Let me know if you would have something more.

**Reply by davehouser1** [reply]
Thanks for the quick response @scholzj!
That documentation you shared makes sense on how to configure storageClasses. However not in a whole, with respect to how to deploy a whole environment fresh. 
The quick start guides seems to direct the user to deploy via direct http manifest links without editing. There is use of sed to adjust namespaces but nothing about how to apply customizations to a fresh deployment. 
Would I need to dig though all files in the install/ directory to apply the storageClass configurations I want? 
Are there any guides / examples I can follow that talk about deploying with customization's from a fresh deployment?

**Reply by scholzj** [reply]
Well, as I said ... Quickstart is a quick start. It cannot cover everything. If you download one of the `strimzi-0.22.1.*` files form the release page (https://github.com/strimzi/strimzi-kafka-operator/releases/tag/0.22.1) it has inside the example. You can just  pick the persistent one, edit it and do `kubectl apply -f` on it for example. It is just as you would work with any other Kubernetes resource. On Linux or MacOS you can for example also do something like `curl -L0 https://strimzi.io/examples/latest/kafka/kafka-persistent.yaml | vim -`, edit it and then just do `:w ! kubectl apply -f -`.

**Reply by davehouser1** [reply]
This was helpful, thank you @scholzj , My cluster deployed with my storageClass.

**Reply by chiheb08** [reply]
@davehouser1 can you please give me a hint on how to do it with a custom storageClass ?

**Reply by riofajars** [reply]
> @davehouser1 can you please give me a hint on how to do it with a custom storageClass ?

you just need to add class: my-storage-class 

![image](https://user-images.githubusercontent.com/72093195/226548648-cf81c31a-b53c-4fd3-950d-0778e92dd87d.png)
